File: BookExamples/DataSetHelper.cs
using System;
using System.Data;
using System.Diagnostics;
using System.Xml;
using System.Runtime.Serialization;

namespace BookExamples
{
	public class DataSetHelper
	{
    private DataSetHelper() {}

    public static void MoveDataTableSchema(DataTable tblSource, DataTable tblDest)
    {
      // Copy the PrimaryKey
      tblDest.PrimaryKey = tblSource.PrimaryKey;

      // Add the columns (not copies)
      foreach(DataColumn col in tblSource.Columns)
      {
        tblSource.Columns.Remove(col.ColumnName);
        tblDest.Columns.Add(col);
      }

      // Add the Constraints
      foreach(Constraint con in tblSource.Constraints)
      {
        tblDest.Constraints.Add(con);
      }

      // Add the PrimaryKey
      tblDest.PrimaryKey = tblSource.PrimaryKey;

      // Add the Relationships
      foreach(DataRelation rel in tblSource.ChildRelations)
      {
        tblDest.ChildRelations.Add(rel);
      }
      foreach(DataRelation rel in tblSource.ParentRelations)
      {
        tblDest.ParentRelations.Add(rel);
      }

      // Extended Properties
      tblDest.TableName = tblSource.TableName;
    }
	}
}

File: BookExamplesVB/DataSetHelper.vb
Imports System
Imports System.Data
Imports System.Diagnostics
Imports System.Xml
Imports System.Runtime.Serialization


Namespace BookExamples
    _
  Public Class DataSetHelper

    Private Sub New()
    End Sub 'New

    Public Shared Sub MoveDataTableSchema(ByVal tblSource As DataTable, ByVal tblDest As DataTable)
      ' Copy the PrimaryKey
      tblDest.PrimaryKey = tblSource.PrimaryKey

      ' Add the columns (not copies)
      Dim col As DataColumn
      For Each col In tblSource.Columns
        tblSource.Columns.Remove(col.ColumnName)
        tblDest.Columns.Add(col)
      Next col

      ' Add the Constraints
      Dim con As Constraint
      For Each con In tblSource.Constraints
        tblDest.Constraints.Add(con)
      Next con

      ' Add the PrimaryKey
      tblDest.PrimaryKey = tblSource.PrimaryKey

      ' Add the Relationships
      Dim rel As DataRelation
      For Each rel In tblSource.ChildRelations
        tblDest.ChildRelations.Add(rel)
      Next rel

      For Each rel In tblSource.ParentRelations
        tblDest.ParentRelations.Add(rel)
      Next rel

      ' Extended Properties
      tblDest.TableName = tblSource.TableName
    End Sub 'MoveDataTableSchema
  End Class 'DataSetHelper
End Namespace 'BookExamples